express-api/src/services/parcels/parcelsService.ts

```diff
@@ -0,0 +1,279 @@
+  });
+  return result;
+};
+
+const updateParcel = async (
+  parcel: Parcels,
+  user: KeycloakUser & KeycloakIdirUser,
+  newSubdivisionPids?: string[],
+) => {
+  // TODO: Does user have matching agency?
+  // Get original parcel
+  const original = await parcelsRepository.findOneByOrFail({ Id: parcel.Id }).catch(() => {
+    throw new ErrorWithCode(`Parcel ID ${parcel.Id} not found`, 404);
+  });
+  // TODO: Is this property in a project already? If so, reject update.
+
+  const classificationTypes = await AppDataSource.getRepository(PropertyClassifications).find();
+  // Unless admin...
+  if (!hasRole(user, [Roles.ADMIN])) {
+    // no switching of agencies
+    if (original.Agency != parcel.Agency) {
+      throw new ErrorWithCode(`Parcel cannot be transferred to the specified agency.`, 403);
+    }
+    // No disposing of property
+    if (
+      parcel.Classification.Id ===
+      classificationTypes.find((classification) => classification.Name === 'Disposed').Id
+    ) {
+      throw new ErrorWithCode(`Parcel classification cannot be changed to Disposed.`, 403);
+    }
+
+    // No changing classification to subdivided
+    if (
+      parcel.Classification.Id ===
+      classificationTypes.find((classification) => classification.Name === 'Subdivided').Id
+    ) {
+      throw new ErrorWithCode(`Parcel classification cannot be changed to Subdivided.`, 403);
+    }
+  }
+
+  // No making properties visible through update
+  if (original.IsVisibleToOtherAgencies != parcel.IsVisibleToOtherAgencies) {
+    throw new ErrorWithCode(
+      `Cannot make a parcel visible to other agencies through this service.`,
+      403,
+    );
+  }
+  // No changing parcels to demolished classification
+  if (
+    parcel.Classification.Id ===
+    classificationTypes.find((classification) => classification.Name === 'Demolished').Id
+  ) {
+    throw new ErrorWithCode(`Parcel classification cannot be changed to Demolished.`, 403);
+  }
+
+  // TODO: Do we need to check buildings, or will they be included in this update?
+
+  // Update parcel
+  const updatedParcel = await parcelsRepository.update({ Id: parcel.Id }, parcel);
+
+  // If there are any attached subdivisions, update them with the new PID
+  newSubdivisionPids.forEach(async (pid) => {
+    await parcelsRepository.update({ PID: pidStringToNumber(pid) }, { ParentParcel: parcel });
+  });
+ 
+  // TODO: Add/Update Fiscals and Evaluations
+  
+  // TODO: Remove outdated fiscals and evaluations
+
+  // TODO: Do we need to verify buildings? Do we remove them if not in incoming parcel?
+  
+  return updatedParcel;
+};
+
+const updateParcelFinancials = async () => {
+  // TODO: Does user have matching agency?
+};
+
+const deleteParcel = async (id: number) => {
+  // TODO: Does user have permission? And agency?
+  // TODO: Is this property in a project already? If so, reject deletion.
+
+  // TODO: Do we delete related buildings too?
+
```

Comment:
Also, do we delete associated records in the parcel evaluations and parcel fiscals as well?
Might need to have a check for subdivisions?